### DISCUSSION TITLE
Cache Control headers when `revAssetUrls` is enabled

### DISCUSSION THREAD

**Discussion by lukasIO** [craftcms/cms#9604] [Ideas] [answered]
Hi,

since 3.7 there is the new config option `revAssetUrls` which we were looking forward to already!

This setting would allow, in my understanding, to set cache control headers on the asset urls with a very high `max-age` duration and (additionally) the `immutable` directive:

Once the `revAssetUrls` setting is enabled each modification of that asset will have its own unique URL, which would allow to skip server validation requests for that asset.

**Comment by andris-sevcenko** [comment, marked as answer]
Wouldn't setting the headers be up to the webserver, not Craft? Even in the AWS S3 scenario, the headers to send along must be provided when the Asset is uploaded, not when it's downloaded.

**Reply by lukasIO** [reply]
You are right, in circumstances where you have control over the webserver, this is probably preferable.
However in other situations (e.g. shared hosting environments) you rarely have fine grained control over the headers being set.
While typing this out, I'm realizing that in those shared hosting environments it would still be possible to place custom `.htaccess` files in the local volume folders that add those headers...

**Reply by andris-sevcenko** [reply]
Right, what I was trying to say was that I wasn't sure if there was anything that Craft could really do about those headers.

**Reply by lukasIO** [reply]
I'm not familiar with how the asset access is handled, but if it is handled through yii, those headers could be added to the response afaik
https://www.yiiframework.com/doc/guide/2.0/en/runtime-responses#http-headers

**Reply by andris-sevcenko** [reply]
It is not handled via Yii or Craft. When an URL to the Asset (revved or not) is generetad, that URL bypasses the CMS.

**Reply by lukasIO** [reply]
makes sense, thanks for clarifying!

**Reply by andris-sevcenko** [reply]
Sorry there wasn't an elegant solution to this!
